js/huobi.js

```diff
@@ -1287,36 +1285,45 @@ module.exports = class huobi extends Exchange {
                 'quoteId': quoteId,
                 'settleId': settleId,
                 'type': type,
-                'contract': contract,
                 'spot': spot,
-                'future': future,
+                'margin': (spot && hasLeverage),
                 'swap': swap,
+                'future': future,
+                'option': false,
+                'active': active,
+                'contract': contract,
                 'linear': linear,
                 'inverse': inverse,
-                'expiry': expiry,
-                'expiryDatetime': this.iso8601 (expiry),
-                'contractSize': contractSize,
-                'active': active,
-                'precision': precision,
                 'taker': taker,
                 'maker': maker,
+                'contractSize': contractSize,
+                'expiry': expiry,
+                'expiryDatetime': this.iso8601 (expiry),
+                'strike': undefined,
+                'optionType': undefined,
+                'precision': {
+                    'amount': amountPrecision,
+                    'price': pricePrecision,
+                    'cost': costPrecision,
+                },
                 'limits': {
+                    'leverage': {
+                        'min': this.parseNumber ('1'),
```

Comment:
We should probably switch to their new v2 endpoint for spot symbols.
In general, for spot markets, the leverage must be undefined, unless the symbol is eligible for margin trading according to their docs. The information on whether it's eligible for margin trading or not is only available from the v2 spot symbols endpoint, afaik.

According to the exchange's API docs, they added this new endpoint in **february** 2022, heh:

<img width="835" alt="Screenshot 2022-01-28 at 20 39 00" src="https://user-images.githubusercontent.com/1294454/151595235-f51f7081-6b7a-4549-99bf-406ddc92ae2c.png">

Reply:
Will need to update their API definitions for those new endpoints that were added this month.

Reply:
Updating the paths should be part of a different PR, I've tried updating paths for other exchanges and the work ends up spidering. 

I don't think that the leverage should be undefined or a number depending on the market. It forces dealing with multiple data types, it's more complicated for our code and for the users. I don't think there's any confusion is giving a max leverage of 1

Also, a maxLeverage of undefined can have a double meaning, either the market has no leverage, or the leverage isn't available within the fetchMarkets method